Compilation Error

Hey, not sure if you still want to mess with this project, but I would find it really useful to translate arbitrary CHCs to VMT. 

Currently I'm getting these 2 errors when I try to compile. For reference I'm using Z3 4.8.16 and LLVM 10.0.0.

```
/Users/colevick/Documents/cs/research/horn2vmt/tools/horn2vmt/main.cc:1622:27: error: loop variable 'input' of type 'const z3::expr &&' binds to a temporary constructed from type '(anonymous namespace)::TransitionSystem::ConstInputIterator::reference' (aka 'const (anonymous namespace)::PrimaryInput &') [-Werror,-Wrange-loop-construct]
    for (const z3::expr&& input : boost::make_iterator_range(xsys_.ibegin(), xsys_.iend())) {
                          ^
/Users/colevick/Documents/cs/research/horn2vmt/tools/horn2vmt/main.cc:1622:10: note: use non-reference type 'z3::expr' to make construction explicit or type 'const (anonymous namespace)::PrimaryInput &' to prevent copying
    for (const z3::expr&& input : boost::make_iterator_range(xsys_.ibegin(), xsys_.iend())) {
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/colevick/Documents/cs/research/horn2vmt/tools/horn2vmt/main.cc:1819:41: error: loop variable 'entry' of type 'std::pair<z3::sort, int> &&' binds to a temporary constructed from type 'std::__hash_map_iterator<std::__hash_iterator<std::__hash_node<std::__hash_value_type<z3::sort, int>, void *> *>>::reference' (aka 'pair<const z3::sort, int> &') [-Werror,-Wrange-loop-construct]
        for (std::pair<z3::sort, int>&& entry : counts) {
                                        ^
/Users/colevick/Documents/cs/research/horn2vmt/tools/horn2vmt/main.cc:1819:14: note: use non-reference type 'std::pair<z3::sort, int>' to make construction explicit or type 'const std::__hash_map_iterator<std::__hash_iterator<std::__hash_node<std::__hash_value_type<z3::sort, int>, void *> *>>::value_type &' (aka 'const pair<const z3::sort, int> &') to prevent copying
        for (std::pair<z3::sort, int>&& entry : counts) {
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,7 @@ if (COMPILE_WITH_COVERAGE)
   endif()
 endif()
 
-add_compile_options(-Wall -Wextra -Werror -Wno-unused-function -Wunused-variable
+add_compile_options(-Wall -Wextra -Wno-unused-function -Wunused-variable
   -Wno-reorder -fno-omit-frame-pointer -g)
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
   add_compile_options( -Weffc++ -Wheader-hygiene -Wold-style-cast)

--- a/default.nix
+++ b/default.nix
@@ -12,8 +12,8 @@
     src = fetchFromGitHub {
       owner = "dbueno";
       repo = "${pname}";
-      rev = "dev";
-      sha256 = "1i4bpvican8v6l5xw25m52n3fg45srwbaqa32z9izvw8v4naviw5";
+      rev = "37b90f62275a24ee686071ef0ec5879374286f99";
+      sha256 = "sha256-4TliZodlnWX7VxIh4ZLbU2XpmVGTSvhLnbQ5//yCpyE=";
     };
 
     fmt = stdenv.mkDerivation {

--- a/flake.nix
+++ b/flake.nix
@@ -24,6 +24,7 @@
             packages = with pkgs; [ creduce ];
             inherit (horn2vmt) cmakeFlags;
             hardeningDisable = [ "all" ];
+            CXXFLAGS = "-Werror";
           };
       });
 }

--- a/flake.nix
+++ b/flake.nix
@@ -1,0 +1,29 @@
+{
+  description = "A very basic flake";
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, flake-compat }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+        horn2vmt = pkgs.callPackage ./default.nix {};
+      in {
+        packages = {
+          inherit horn2vmt;
+        };
+        defaultPackage = horn2vmt;
+        devShell =
+          pkgs.mkShell {
+            inputsFrom = [ horn2vmt ];
+            packages = with pkgs; [ creduce ];
+            inherit (horn2vmt) cmakeFlags;
+            hardeningDisable = [ "all" ];
+          };
+      });
+}

--- a/tools/horn2vmt/main.cc
+++ b/tools/horn2vmt/main.cc
@@ -1619,7 +1619,7 @@ class Horn2Vmt {
       ++index;
     }
     fmt::print("; {} inputs\n", xsys_.num_inputs());
-    for (const z3::expr&& input : boost::make_iterator_range(xsys_.ibegin(), xsys_.iend())) {
+    for (z3::expr input : boost::make_iterator_range(xsys_.ibegin(), xsys_.iend())) {
       fmt::print("{}\n", input.decl());
     }
     fmt::print("(define-fun .def{} () Bool (! {} :init true))\n", index++,
@@ -1816,7 +1816,7 @@ void Horn2Vmt::CreateStateSpace() {
           }
           counts[sort]++;
         }
-        for (std::pair<z3::sort, int>&& entry : counts) {
+        for (std::pair<z3::sort, int> entry : counts) {
           auto sort = entry.first;
           auto num_occurrences = entry.second;
           int num_places =

